### PR TITLE
Print loglines from chaosduck to all tests.

### PR DESCRIPTION
**Pull request by markusthoemmes**
To enhance debuggability in flaky tests that are suspicious to be chaos induced.

/assign @vagababov @mattmoor

### CODE REVIEW

**Comment by markusthoemmes** [comment]
Verified to work in https://github.com/knative/serving/pull/8890

```
    kubelogs.go:218: I 13:28:36.535 chaosduck-cb6bb878d-rnvkz 2020/08/03 13:28:36 Quacking at "net-certmanager-webhook" leader "net-certmanager-webhook-67b9dcc658-4bsqw"
```

File: test/logstream/kubelogs.go [hunk 1]
```diff
@@ -201,6 +207,18 @@ func (k *kubelogs) handleLine(l []byte, pod string) {
...
 
+// handleGenericLine prints the given logline to all active tests as it cannot be parsed
+// and/or doesn't contain any correlation data (like the chaosduck for example).
+func (k *kubelogs) handleGenericLine(l []byte, pod string) {
+	k.m.RLock()
+	defer k.m.RUnlock()
+
+	for _, logf := range k.keys {
+		// I 15:04:05.000 webhook-699b7b668d-9smk2 this is my message
+		logf(fmt.Sprintf("I %s %s %s", time.Now().Format(timeFormat), pod, string(l)))
```
Comment (vagababov) [inline test/logstream/kubelogs.go:218]: Just format using logf.

File: test/logstream/kubelogs.go [hunk 2]
```diff
@@ -77,7 +77,13 @@ func (k *kubelogs) startForPod(eg *errgroup.Group, pod *corev1.Pod) {
...
 			scanner := bufio.NewScanner(stream)
 			for scanner.Scan() {
-				k.handleLine(scanner.Bytes(), pn)
+				// Specialcase logs from chaosduck to be able to easily see when pods
+				// have been killed throughout all tests.
+				if cn == "chaosduck" {
+					k.handleGenericLine(scanner.Bytes(), pn)
+				} else {
+					k.handleLine(scanner.Bytes(), pn)
+				}
```
Comment (vagababov) [inline test/logstream/kubelogs.go:86]: given the amount of logs processing we're doing — may be just pass the processor func to the function
and check for the type of binary once?

**Pull request merged**